core/ast2100/ast2100.js

```diff
@@ -0,0 +1,516 @@
+    var Xusock = typeof UI !== "undefined" ? UI.rfb._sock : rfb._sock;
+    RFB.messages.atenChangeVideoSettings(sock, lumaQt, chromaQt, subsamplingMode);
+};
+
+
+var Ast2100Decoder;
+
+
+(function () {
+    "use strict";
+    
+    Ast2100Decoder = function (defaults) {
+
+        this._blitCallback = defaults.blitCallback;
+        this._videoSettingsChangedCallback = defaults.videoSettingsChangedCallback;
+        this._frame_width = defaults.width;
+        this._frame_height = defaults.height;
+
+        if (!this._frame_width || !this._frame_height)
+            throw 'Missing required parameter: width, height';
+
+        // Either 444u or 422u (though the "422" mode is really 4:2:0, where
+        // chroma is subsampled by a factor of two in each direction)..  Applies
+        // only to JPEG-ish blocks, not to VQ blocks; but VQ blocks seem to only
+        // appear in 4:4:4 color mode.
+        this.subsamplingMode = -1;
+        
+        this.quantTables = [new Int32Array(64), new Int32Array(64)];
+        this._loadedQuantTables = [-1, -1];
+
+        this.huffTables = [
+            [new JpegHuffmanTable({bits: BITS_DC_LUMA, huffval: HUFFVAL_DC_LUMA}),
+             new JpegHuffmanTable({bits: BITS_DC_CHROMA, huffval: HUFFVAL_DC_CHROMA})],
+            [new JpegHuffmanTable({bits: BITS_AC_LUMA, huffval: HUFFVAL_AC_LUMA}),
+             new JpegHuffmanTable({bits: BITS_AC_CHROMA, huffval: HUFFVAL_AC_CHROMA})],
+        ];
+        
+        this._scan_components = [
+            {huffTableSelectorDC: 0, huffTableSelectorAC: 0},  // Y
+            {huffTableSelectorDC: 1, huffTableSelectorAC: 1},  // Cb
+            {huffTableSelectorDC: 1, huffTableSelectorAC: 1}   // Cr
+        ];
+        this._scan_prev_dc = [0, 0, 0];
+
+        this._mcuPosX = 0;
+        this._mcuPosY = 0;
+
+        this._initializeVq();
+
+        // Allocate memory here once and then re-use it.  These buffers store
+        // data after entropy decoding (Huffman, zigzag) and before we dequant
+        // and apply the IDCT.
+        this._tmpBufY = [
+            new Int16Array(DCTSIZE2),
+            new Int16Array(DCTSIZE2),
+            new Int16Array(DCTSIZE2),
+            new Int16Array(DCTSIZE2)];
+        this._tmpBufCb = new Int16Array(DCTSIZE2);
+        this._tmpBufCr = new Int16Array(DCTSIZE2);
+
+        // These buffers store IDCT output.
+        this._componentBufY = [
+            new Uint8Array(DCTSIZE2),
+            new Uint8Array(DCTSIZE2),
+            new Uint8Array(DCTSIZE2),
+            new Uint8Array(DCTSIZE2)];
+        this._componentBufCb = new Uint8Array(DCTSIZE2);
+        this._componentBufCr = new Uint8Array(DCTSIZE2);
+        
+        // The final RGB output goes here; noVNC expects each group of four
+        // elements to represent one RGBA pixel.  this._outputBuf = new
+        // Uint8Array(DCTSIZE2 * 4);
+        this._outputBuf = new Uint8Array(DCTSIZE2 * 4);
+        
+    };
```

Comment:
Unfortunately, no.  You're correct that some data is received in a format that looks like JPEG if you squint, but individual blocks that look much like JPEG MCUs can be interleaved with blocks compressed by vector quantization.

This should be covered in the cleaned-up version of my notes on the ins and outs of the ATEN RFB implementation and those on the 0x57 encoding.